src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1308,53 +1275,19 @@ impl LinkCollector<'_, '_> {
         &mut self,
         key: ResolutionInfo,
         diag: DiagnosticInfo<'_>,
-        cache_resolution_failure: bool,
-    ) -> Option<(Res, Option<UrlFragment>)> {
-        if let Some(ref cached) = self.visited_links.get(&key) {
-            match cached {
-                Some(cached) => {
-                    return Some(cached.res.clone());
-                }
-                None if cache_resolution_failure => return None,
-                None => {
-                    // Although we hit the cache and found a resolution error, this link isn't
-                    // supposed to cache those. Run link resolution again to emit the expected
-                    // resolution error.
```

Comment:
Maybe keep some form of this comment? (It could be rewritten or trimmed.)

Reply:
It was kept, just moved to the call site of `resolve_with_disambiguator_cached`.